src/language-js/needs-parens.js

```diff
@@ -34,7 +34,7 @@ function hasClosureCompilerTypeCastComment(text, path, locStart, locEnd) {
         comment =>
           comment.leading &&
           comments.isBlockComment(comment) &&
-          comment.value.match(/^\*\s*@type\s*{[^}]+}\s*$/) &&
+          comment.value.match(/^\*\s*@type\s*{(.|\n)+}\s*$/) &&
```

Comment:
This also matches 

```
* @type {A} 
* @something {B}
```

which does not look right. It should be better to write a simple function to match nested brackets.